app/src/main/java/ch/heigvd/pro/b04/android/Poll/Question/QuestionAdapter.java

```diff
@@ -0,0 +1,106 @@
+package ch.heigvd.pro.b04.android.Poll.Question;
+
+import android.graphics.Color;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+import android.widget.Button;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LifecycleOwner;
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import ch.heigvd.pro.b04.android.Poll.PollViewModel;
+import ch.heigvd.pro.b04.android.R;
+
+public class QuestionAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder>  {
+    private static final int VIEW_TYPE_HEADER = 0;
+    private static final int VIEW_TYPE_QUESTION = 1;
+
+    private PollViewModel state;
+    private Set<Integer> answered = new HashSet<>();
+
+    public QuestionAdapter(PollViewModel state, LifecycleOwner lifecycleOwner) {
+        this.state = state;
+
+        state.getAnsweredQuestion().observe(lifecycleOwner, questions -> {
+            answered.clear();
+            for (Question question : questions) {
+                answered.add(question.getId() + 1);
```

Comment:
I'm not so sure about the principle of saving positions that have been answered, rather than the list of questions directly. Maybe the state of whether a question is answered or not could be stored in the `Question` directly, and we would be provided with a `List<Question>` to display rather than adding some logic in the  `RecyclerView.Adapter` ?